include/zephyr/net/ethernet.h

```diff
@@ -1287,6 +1291,85 @@ static inline bool net_eth_is_vlan_interface(struct net_if *iface)
 #define ETH_NET_L3_REGISTER(name, ptype, handler) \
 	NET_L3_REGISTER(&NET_L2_GET_NAME(ETHERNET), name, ptype, handler)
 
+enum net_eth_mac_type {
```

Comment:
I think this deserves an inline description these types?

Reply:
Doxygen bits added.